### DISCUSSION TITLE
Configure default_queue_type rabbitmq 3.13.4 but not working

### DISCUSSION THREAD

**Discussion by chungtd203338** [rabbitmq/rabbitmq-server#12077] [Questions] [answered]
I configured default_queue_type to configure the default queue type on all vhosts in the rabbitmq node as quorum, however when creating a new vhost, it is still in not set status.

![abc](https://github.com/user-attachments/assets/98f70622-dbfd-4d72-9585-39456b0a0243)

I checked the configuration in rabbitmq nodes with the command rabbitmq environment or cat /etc/rabbitmq/rabbitmq.conf both are there but it doesn't seem to work.

**Comment by michaelklishin** [comment]
We do not have enough information to suggest anything. We do not guess in this community, so please put together an executable way to reproduce.

`default_queue_type` in `rabbitmq.conf` has the lowest priority, lower than [the same setting for virtual hosts](https://www.rabbitmq.com/docs/vhosts#default-queue-type) and lower than the client-provided value. Other than this, I cannot suggest anything with a single screenshot for "proof."

**Reply by chungtd203338** [reply]
I am implementing rabbitmq in cluster model on K8s environment, I want my queues to be scalable and fault tolerant by using quorum queue. I want users to not need to care what type of queue is being used and just need to declare the queue usage, it will default to quorum queue.

I see that from rabbitmq version 3.13.4 there is support to configure **_default_queue_type_** for all vhosts in the cluster but when configuring it it doesn't work properly.

Here is rabbitmq.conf file

![image](https://github.com/user-attachments/assets/e24155be-ca38-4b64-8d7c-37ebc3cd380c)

![image](https://github.com/user-attachments/assets/050e64cd-340b-46c4-9f40-76b3fdfb9a67)

I made a vhost declaration without specifying default_queue_type

![image](https://github.com/user-attachments/assets/95145829-fe03-4ae5-b098-8ebe2f665a64)

![image](https://github.com/user-attachments/assets/d64f8dbe-b412-4d1c-85af-cbc566d39687)

This is my problem, i still don't know how to handle it :(

**Reply by michaelklishin** [reply]
Apparently for virtual hosts that do not have any metadata at all (no limits, no DQT), the value has no effect. I suspect that every test environment does have virtual host limits because it was contributed by a public cloud infrastructure provider.

I will take a look at how quickly this can be added. For now (and arguably in general), use the [per-virtual host DQT](https://www.rabbitmq.com/docs/vhosts#default-queue-type).

**Reply by michaelklishin** [reply, marked as answer]
https://github.com/rabbitmq/rabbitmq-server/pull/12109 should address this. It will ship in 4.0.0. There won't be any more open source `3.13.x` releases, so this change won't be backported.

Configuring a DQT and any other virtual metadata key, such as a description or a very high limit, can be used as a workaround with `3.13.7`. As well as [configuring DQT for individual virtual hosts](https://www.rabbitmq.com/docs/vhosts#default-queue-type).

**Reply by jonenst** [reply]
I got bit by this on 3.13.7 for a toy deployment on my computer, global DQT from rabbitmq.conf was ignored. worked on 4.0.2.

**Reply by yuvalgl** [reply]
Hi,

any update on this feature for 3.13.x?

it would really help if we could finish a full migration to quorum queues before V4.

**Reply by michaelklishin** [reply]
@yuvalgl there won't be any more open source RabbitMQ 3.13.x releases. This issue does not affect `4.0.x` releases. [DQT](https://www.rabbitmq.com/docs/vhosts#default-queue-type) at the virtual host level is far from being the only or the primary way of moving to quorum queues.

Configuring a DQT and any other metadata key per virtual host, such as a description or a very high limit applications will never surpass, can be used as a workaround with `3.13.7`.
